Library/ViewModels/SettingsRequestDataCellViewModel.swift

```diff
@@ -0,0 +1,145 @@
+  var goToSafari: Signal<String, NoError> { get }
+  var requestDataButtonEnabled: Signal<Bool, NoError> { get }
+  var requestedDataExpirationDate: Signal<String, NoError> { get }
+  var requestDataLoadingIndicator: Signal<Bool, NoError> { get }
+  var requestDataText: Signal<String, NoError> { get }
+  var requestDataTextHidden: Signal<Bool, NoError> { get }
+  var showPreparingDataText: Signal<Bool, NoError> { get }
+  var showRequestDataPrompt: Signal<(), NoError> { get }
+  var unableToRequestDataError: Signal<String, NoError> { get }
+}
+
+public protocol SettingsRequestDataCellViewModelType {
+  var inputs: SettingsRequestDataCellViewModelInputs { get }
+  var outputs: SettingsRequestDataCellViewModelOutputs { get }
+}
+
+public final class SettingsRequestDataCellViewModel: SettingsRequestDataCellViewModelType,
+  SettingsRequestDataCellViewModelInputs, SettingsRequestDataCellViewModelOutputs {
+
+  public init() {
+    let initialUser = self.configureWithUserProperty.signal
+      .skipNil()
+
+    let emptyStringOnLoad = self.configureWithUserProperty.signal.mapConst("")
+
+    let exportEnvelope = initialUser
+      .switchMap { _ in
+        AppEnvironment.current.apiService.exportDataState()
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .demoteErrors()
+    }
+
+    self.showRequestDataPrompt = exportEnvelope
+      .takeWhen(self.exportDataTappedProperty.signal.ignoreValues())
+      .filter { $0.dataUrl == nil || $0.state == .expired || $0.expiresAt == nil }
+      .ignoreValues()
+
+    let requestDataEvent = self.startRequestDataTappedProperty.signal
+      .switchMap { _ in
+        AppEnvironment.current.apiService.exportData()
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .materialize()
+    }
+
+    self.unableToRequestDataError = requestDataEvent.errors()
+      .map { env in
+        env.errorMessages.first ?? "unable to request data"
+    }
+
+    self.requestDataLoadingIndicator = Signal.merge(
+      self.configureWithUserProperty.signal.mapConst(false),
+      exportEnvelope.map { $0.state == .processing },
+      self.startRequestDataTappedProperty.signal.mapConst(true)
+    )
+
+    self.requestDataText = Signal.merge(
+     emptyStringOnLoad,
+     exportEnvelope
+      .map { $0.state == .expired || $0.expiresAt == nil || $0.dataUrl == nil
+        ? Strings.Request_my_Personal_Data() : Strings.Download_your_personal_data() }
+    )
+
+    self.requestDataButtonEnabled = self.requestDataLoadingIndicator.signal
+      .map { !$0 }
+
+    self.requestedDataExpirationDate = Signal.merge(
+      emptyStringOnLoad,
+      exportEnvelope.map {
+        dateFormatter(for: $0.expiresAt, state: $0.state)
+      }
+    )
+
+    self.dataExpirationAndChevronHidden = exportEnvelope
+      .map { $0.state == .expired || $0.expiresAt == nil || $0.dataUrl == nil ? true : false }
```

Comment:
This logic: `$0.state == .expired || $0.expiresAt == nil || $0.dataUrl == nil` is used in a couple places, would it make sense to put it in it's own function?